### PR TITLE
[Loader] Stop attempting to load Fabric mods

### PR DESCRIPTION
**Pull request by ChachyDev**
This pull requests stops discovering Fabric mods. 

If I'm honest I don't have a strong feeling this will be accepted.

Note: The diff may be bigger due to my habbit of reformatting when coding.

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java [hunk 1]
```diff
@@ -52,124 +51,102 @@ public void findClasspathMods(ModClassLoader modClassLoader)
...
+                    ZipEntry entry = zipFile.getEntry("fabric.mod.json");
+                    zipFile.close();
+                    if (entry != null) {
+                        FMLLog.info("It looks like " + modFile.getName() + " is a Fabric Mod? Skipping");
+                        continue;
+                    }
+                } catch (IOException exception) {
+                    exception.printStackTrace();
+                }
+            }
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java:98]: This is what you'll wnat to be reviewing :)

File: src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java [hunk 2]
```diff
@@ -52,124 +51,102 @@ public void findClasspathMods(ModClassLoader modClassLoader)
...
         modList = FileListHelper.sortFileList(ObjectArrays.concat(modList, supplementalModFileCandidates, File.class));
-        for (File modFile : modList)
-        {
+        for (File modFile : modList) {
+            if (modFile.getName().endsWith(".jar") || modFile.getName().endsWith(".zip")) {
+                try {
+                    // Skip Fabric mods
+                    ZipFile zipFile = new ZipFile(modFile);
+                    ZipEntry entry = zipFile.getEntry("fabric.mod.json");
+                    zipFile.close();
```
Comment (asbyth) [inline src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java:90]: `try (ZipFile zipFile = ...)`?

File: src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java [hunk 3]
```diff
@@ -52,124 +51,102 @@ public void findClasspathMods(ModClassLoader modClassLoader)
...
+                    if (entry != null) {
+                        FMLLog.info("It looks like " + modFile.getName() + " is a Fabric Mod? Skipping");
```
Comment (asbyth) [inline src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java:92]: "{} may be a non-Forge mod! Skipping for now." seems to be better wording?
Maybe add a config setting for this? I'm not sure where this is done as I've yet to modify their config at any point in time, but you'll figure it out.

File: src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java (hunk 1 above)
Comment (asbyth) [reply src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java:98]: I'll eventually get around to reformatting the entire project as their C# style sucks lol

**Review by asbyth — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/common/ForgeModContainer.java [hunk 4]
```diff
@@ -180,19 +161,23 @@ private static void syncConfig(boolean load)
...
+        if (removeErroringEntities) {
             FMLLog.warning("Enabling removal of erroring Entities - USE AT YOUR OWN RISK");
         }
 
+        prop = config.get(CATEGORY_GENERAL, "stopDiscoveringNonForgeMods", false);
+        prop.comment = "Stops discovering Non-Forge (Fabric etc.) mods";
+        prop.setLanguageKey("forge.configgui.stopDiscoveringNonForgeMods").setRequiresMcRestart(true);
+        stopDiscoveringNonForgeMods = prop.getBoolean(false);
+        propOrder.add(prop.getName());
+
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/common/ForgeModContainer.java:173]: Looking for this :)

File: src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java [hunk 5]
```diff
@@ -82,15 +83,16 @@ public void findModDirMods(File modsDir, File[] supplementalModFileCandidates) {
...
-                    if (entry != null) {
-                        FMLLog.info("It looks like " + modFile.getName() + " is a Fabric Mod? Skipping");
-                        continue;
+                    try (ZipFile file = new ZipFile(modFile)) {
+                        ZipEntry entry = file.getEntry("fabric.mod.json");
+
+                        if (entry != null) {
+                            FMLLog.info("%s may be a non-Forge mod! Skipping for now.", modFile.getName());
+                            continue;
+                        }
```
Comment (ChachyDev) [inline src/main/java/net/minecraftforge/fml/common/discovery/ModDiscoverer.java]: And this.

**Review by asbyth — APPROVED** [review]

**Pull request merged**